File: resources/scop_1.5.1/java/examples/Sender.java
// Sender.java - DMI - 20-9-02

/* Usage: java Sender [ <message> ]   (default message is "Hello world!") */

import java.io.*;

public class Sender
{
	public static void main(String[] args)
	{
		SCOP scop;
		String msg = args.length > 0 ? args[0] : "Hello world!";
		
		scop = new SCOP("localhost", "Sender");
		scop.send_message("Receiver", msg);
		scop.close();
	}
}

File: resources/scop_1.5.1/examples/sender.cpp
// sender.cpp - DMI - 7-9-02

/* Usage: sender [ <message> ]   (default message is "Hello world!") */

#include <scop.h>

int main(int argc, char **argv)
{
	int sock;
	char *msg = argc > 1 ? argv[1] : (char *)"Hello world!";
	
	sock = scop_open("localhost", "sender");
	scop_send_message(sock, "receiver", msg);
	
	close(sock);
	return 0;
}
